core/domain/feedback_domain.py

```diff
@@ -362,6 +365,55 @@ def to_dict(self) -> FeedbackMessageReferenceDict:
             'message_id': self.message_id
         }
 
+    @classmethod
+    def from_dict(
+        cls, reference_dict: FeedbackMessageReferenceDict
+    ) -> FeedbackMessageReference:
+        """Return a FeedbackMessageReference domain object from a dict.
+
+        Args:
+            reference_dict: dict. The dict representation of a
+                FeedbackMessageReference object.
+
+        Returns:
+            FeedbackMessageReference. The corresponding FeedbackMessageReference
+            domain object.
+        """
+
+        try:
+            feedback_message_reference = cls(
+                reference_dict['entity_type'],
+                reference_dict['entity_id'],
+                reference_dict['thread_id'],
+                reference_dict['message_id']
+            )
+        except utils.ValidationError as error:
```

Comment:
```suggestion
        except Exception as error:
```

Reply:
@vojtechjelinek, I realized that Mypy does not let you test the "except"-branch, since that requires passing a faulty dict. This leads to the coverage test failing since that branch is not covered. Given the use of Mypy, I now removed the try/except and instead rely on Mypy.